### DISCUSSION TITLE
elaborate on reindexArrayWithNumericKeysOtherwiseJavaScriptWillMessWithTheOrder

### DISCUSSION THREAD

**Discussion by mathiasgrimm** [livewire/livewire#3600] [1. Help] [answered]
Hi, I have been having a real bad time because of this method.
Could anybody there provide an example of what will be messed up?

A lot of times I like to keep my collection/arrays indexed by IDs. I don't really trust array indexes to be the same across multiple request and think is going to lead to undesirable results in the future like updating the wrong db record, etc. 
Indexing everything by ID is much safer in my opinion. Also, in addition to that, whenever you index anything by key it loses completely the sorting defined in the component. This is a pretty hard "bug" to debug and many might not even notice during development.

**Comment by jonneroelofs** [comment, marked as answer]
You are in luck, they discuss this in the last episode of the (awesome!) podcast "No Plans To Merge". Minute 68:57 

Also if you look at the git blame/history of this function you will probably find some issues related to why this function does what it does today.

https://open.spotify.com/episode/22ui1ge522V6WYze6e88fO?si=acc33d8689de4a4b
